### ISSUE TITLE
Provide way of uninstalling a plugin (or don't suggest upgrading it)

### ISSUE THREAD

**Issue by arthurlogilab** [grafana/grafana#5714]
* **I'm submitting a ...**
- [ ] Bug report
- [X] Feature request
- [ ] Question / Support request: **Please do not** open a github issue. [Support Options](http://grafana.org/support/)

Please include this information:
- What Grafana version are you using?

ii  grafana                           3.1.1-1470047149      amd64                 Grafana

- What datasource are you using?
- What OS are you running grafana on?

debian

- What did you do?

installed worldping, deactivated it

- What was the expected result?

have the possibility of uninstalling it. 

- What happened instead?

the plugin is still there and I get notifications to upgrade it 

![screenshot from 2016-08-02 14-45-20](https://cloud.githubusercontent.com/assets/445200/17330173/15d00eaa-58c6-11e6-97eb-45788fcc74bc.png)

**Comment by torkelo** [comment]
Have you tried 
`grafana-cli plugins uninstall raintank-worldping-app`

**Comment by arthurlogilab** [comment]
@torkelo ok, that works, the feature request could be that this would be available in the web interface since the install is doable through the web interface.

**Comment by siysco** [comment]
Same issue in 2018. I can't remove apps from hosted instances. I have to create a new account each time I want to remove an app.

**Comment by iraklic** [comment]
I have perhaps a related issue. I am using a docker image and I foolishly installed `grafana-cli plugins install grafana-image-renderer` before I read that it is not supported by the image. Now even if I pull the image from scratch is sees the plugin somewhere and fails to run the container. 

How can I remove that plugin, so the image does not try to implement it?

**Comment by andreasnl** [comment]
> 
> 
> I have perhaps a related issue. I am using a docker image and I foolishly installed `grafana-cli plugins install grafana-image-renderer` before I read that it is not supported by the image. Now even if I pull the image from scratch is sees the plugin somewhere and fails to run the container.
> 
> How can I remove that plugin, so the image does not try to implement it?

I encountered the exact same issue and solved it by following the steps described by Dmitriusan here:

https://stackoverflow.com/questions/32353055/how-to-start-a-stopped-docker-container-with-a-different-command

After gaining access to the container, you'll be able to uninstall any plugins by using the grafana-cli:

`grafana-cli plugins remove grafana-image-renderer`

This will in turn enable you to start grafana through the regular entrypoint by running `/run.sh`.

**Comment by ArunaLakmal** [comment]
I had the similar situation with Kubernetes and I was struggling to bring my pod up and running. I deployed through the helm and removing plugin didn't work out for me. 

I manually edited the deployment like this: 

`spec:
  template:
    spec:
      containers:
      - grafana
        command: ["grafana-cli"]
        args: ["plugins", "remove", "grafana-image-renderer"]`


Then the container failed again with below message:
`Removing plugin: grafana-image-renderer
Error: ✗ plugin does not exist`

Then I removed the updated commands in the deployment and it worked out for me.
